quantize.py

```diff
@@ -488,10 +489,18 @@ def __init__(
 
         assert out_features % 8 == 0, "require out_features % 8 == 0"
         assert in_features % (inner_k_tiles * 16) == 0, "require in_features % (innerKTiles * 16) == 0"
-        self.register_buffer(
-            "weight",
-            torch.empty((out_features // 8, in_features // (inner_k_tiles * 16), 32, inner_k_tiles // 2), dtype=torch.int32)
-        )
+        if device == 'cuda':
+            self.register_buffer(
+                "weight",
+                torch.empty((out_features // 8, in_features // (inner_k_tiles * 16), 32, inner_k_tiles // 2), dtype=torch.int32)
+            )
+        elif device == 'cpu':
+            self.register_buffer(
+                "weight",
+                torch.empty((out_features, in_features // 2), dtype=torch.uint8)
+            )
+        else:
+            raise ValueError(f"Invalid  device {device} needs to be one of [cuda, cpu]")
```

Comment:
Hmmm, so if we're on CPU, the weights from CUDA won't work, and vice versa?  Not super exciting...

Should these parameters be independent of the actual device?  If the best params depend on device, maybe we might have some sort of get_config(my_device) for a preferred layout, but the ability to run the other layout if the config is present?, rather than base on CPU... I might have gotten a CUDA model, and just want to run it.  (We might actually need a reformat int4 anyway?)